src/common.h

```diff
@@ -54,66 +54,81 @@ static inline int msg_recvfrom(int fd, uint8_t* buf, int nbytes, struct sockaddr
 {
 	int ret = 0;
 	socklen_t size = sizeof(struct sockaddr_in);
-    int flags = 0;
+	int flags = 0;
 
-#ifdef  USING_VMA_EXTRA_API
+#ifdef USING_VMA_EXTRA_API
 	int remain_buffer, data_to_copy;
 	uint8_t* start_addrs; 
 	struct vma_packet_t *pkt;
+	ZeroCopyData *z_ptr = g_zeroCopyData[fd];
 
-	if (g_pApp->m_const_params.is_vmazcopyread && g_vma_api) {
+	if (z_ptr && g_pApp->m_const_params.is_vmazcopyread) {
```

Comment:
i can use at() to get z_ptr and an exception will occure if something unxpected happend
ZeroCopyData *z_ptr = g_zeroCopyData.at(fd)